src/test/java/com/mongodb/jdbc/MongoSQLMock.java

```diff
@@ -284,4 +358,191 @@ MongoJsonSchemaResult generateSchema() {
+                        + "                    \"bsonType\": \"regex\""
+                        + "                },"
+                        + "                \"dbPointer\": {"
+                        + "                    \"bsonType\": \"dbPointer\""
+                        + "                },"
+                        + "                \"javascript\": {"
+                        + "                    \"bsonType\": \"javascript\""
+                        + "                },"
+                        + "                \"symbol\": {"
+                        + "                    \"bsonType\": \"symbol\""
+                        + "                },"
+                        + "                \"javascriptWithScope\": {"
+                        + "                    \"bsonType\": \"javascriptWithScope\""
+                        + "                },"
+                        + "                \"int\": {"
+                        + "                    \"bsonType\": \"int\""
+                        + "                },"
+                        + "                \"timestamp\": {"
+                        + "                    \"bsonType\": \"timestamp\""
+                        + "                },"
+                        + "                \"long\": {"
+                        + "                    \"bsonType\": \"long\""
+                        + "                },"
+                        + "                \"decimal\": {"
+                        + "                    \"bsonType\": \"decimal\""
+                        + "                },"
+                        + "                \"minKey\": {"
+                        + "                    \"bsonType\": \"minKey\""
+                        + "                },"
+                        + "                \"maxKey\": {"
+                        + "                    \"bsonType\": \"maxKey\""
+                        + "                }"
+                        + "            },"
+                        + "            \"required\": ["
+                        + "                \"double\", \"string\", \"object\", \"array\", \"binData\", \"undefined\","
+                        + "                \"objectId\", \"bool\", \"date\", \"null\", \"regex\", \"dbPointer\","
+                        + "                \"javascript\", \"symbol\", \"javascriptWithScope\", \"int\","
+                        + "                \"timestamp\", \"long\", \"decimal\", \"minKey\", \"maxKey\""
+                        + "            ]"
+                        + "        }"
+                        + "    },"
+                        + "    \"required\": [\"all\"]"
+                        + "}";
+
+        return MongoDriver.registry
+                .get(MongoJsonSchema.class)
+                .decode(new JsonReader(schema), DecoderContext.builder().build());
+    }
+
+    static BsonDocument generateRowAllTypes() {
+        String doc =
+                ""
```

Comment:
Nit: I think this empty string can be safely removed. Or does it come from Spotless?